src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -33,14 +34,16 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(@RequestParam Authentication authentication) {
+        ShoppingCart cart = shoppingCartService.getByUser(userService.findByEmail(authentication
```

Comment:
let's use `authentication.getName()` instead of `getPrincipal()` and `orElseThrow()` approach for getting value from `Optional` instead of simple `get()`.
```
 User user = userService.findByEmail(authentication.getName())
                .orElseThrow(() -> new NoSuchElement("Invalid email"));
```
fix all similar cases in all controller classes where it needed.